### PR TITLE
SASS-8122 - FE-BE Integration - Property Rentals Income CYA

### PR DESCRIPTION
**Pull request by modivilioglu**

### CODE REVIEW

File: app/models/responses/PropertyPeriodicSubmissionRequest.scala [hunk 1]
```diff
@@ -16,72 +16,101 @@
...
+            ukOtherPropertyIncome,
+            UkOtherPropertyExpenses(
+              premisesRunningCosts = expenses.rentsRatesAndInsurance,
+              repairsAndMaintenance = expenses.repairsAndMaintenanceCosts,
+              financialCosts = expenses.loanInterest,
+              professionalFees = expenses.otherProfessionalFee,
+              costOfServices = expenses.costsOfServicesProvided,
+              travelCosts = expenses.propertyBusinessTravelCost,
+              other = expenses.otherAllowablePropertyExpenses,
+              residentialFinancialCostsCarriedForward = None,
```
Comment (arunkumar461) [inline app/models/responses/PropertyPeriodicSubmissionRequest.scala:65]: We shouldn't set any value None,  
reason: There could be chance people might use external software to update these fields so lets not override with None

File: app/controllers/JourneyAnswersController.scala [hunk 2]
```diff
@@ -79,11 +81,13 @@ class JourneyAnswersController @Inject()(propertyService: PropertyService,
       withJourneyContextAndEntity[Expenses](taxYear, incomeSourceId, nino, JourneyName.RentalExpenses, request) { (_, expenses) =>
         handleResponse(CREATED) {
           for {
+            psr <- propertyService.getCurrentPeriodicSubmission(taxYear.endYear, nino.value, nino.value, incomeSourceId.value) //.getPropertyPeriodicSubmissions(taxYear, incomeSourceId)
```
Comment (arunkumar461) [inline app/controllers/JourneyAnswersController.scala:84]: Move this logic to service layer and split the responsibility

**Review by arunkumar461 — CHANGES_REQUESTED** [review]

**Review by vivekhmrc — APPROVED** [review]

File: app/controllers/JourneyAnswersController.scala (hunk 2 above)
Comment (modivilioglu) [reply app/controllers/JourneyAnswersController.scala]: Done.

File: app/models/responses/PropertyPeriodicSubmissionRequest.scala (hunk 1 above)
Comment (modivilioglu) [reply app/models/responses/PropertyPeriodicSubmissionRequest.scala]: Done.

**Pull request merged**